internal/pkg/numalign/numalign.go

```diff
@@ -0,0 +1,147 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ */
+
+package numalign
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"path/filepath"
+	"reflect"
+	"strings"
+
+	"github.com/openshift-kni/debug-tools/internal/pkg/vfs"
+)
+
+const (
+	SysDevicesSystemNodeDir = "devices/system/node"
+	SysBusPCIDevicesDir     = "bus/pci/devices/"
+)
+
+type Resources struct {
+	CPUToNUMANode     map[int]int    `json:"cpus"`
+	PCIDevsToNUMANode map[string]int `json:"pcidevices"`
+}
+
+type Result struct {
+	Aligned    bool       `json:"aligned"`
+	NUMACellID int        `json:"numacellid"`
+	Resources  *Resources `json:"resources",omitempty`
+}
+
+func (re Result) JSON() string {
+	var b strings.Builder
+	enc := json.NewEncoder(&b)
+	enc.Encode(re)
+	return b.String()
+}
+
+func (numaRes *Resources) CheckAlignment() Result {
+	ret := Result{
+		Aligned:    false,
+		NUMACellID: -1,
+		Resources:  numaRes,
+	}
+	for _, cpuNode := range numaRes.CPUToNUMANode {
+		if ret.NUMACellID == -1 {
+			ret.NUMACellID = cpuNode
+		} else if ret.NUMACellID != cpuNode {
+			return ret
+		}
+	}
+	for _, devNode := range numaRes.PCIDevsToNUMANode {
+		if devNode != -1 && ret.NUMACellID != devNode {
+			return ret
+		}
+	}
+	ret.Aligned = true
+	return ret
+}
+
+func (numaRes *Resources) MakeValidationScript() string {
```

Comment:
I do not have a strong opinion here, but out of curiosity what can be a use case that you will want to replace the validation script?

Reply:
let me kill this half-baked feature until we have a clear use case.